Hack in rudimentary string support

jbush001 opened the pull request.

jbush001 committed:
Hack in rudimentary string support

rollbear commented:
One more thing. It's of course up to you, but you may want to add your name in the comment section in the top of basic.hpp

jbush001 commented:
I've updated the pull request to include your feedback
- Added a common type_mismatch function
- Updated example to use a string (append dots to DEPTH)
- Updated README to remove message that strings aren't supported

rollbear commented:
Done. Thanks!
